lib/magma/unloader.rb

```diff
@@ -0,0 +1,30 @@
+require_relative './tsv_writer'
+require_relative './retrieval'
+
+class ModelUnloader
+
+  def initialize(model, payload = Magma::Payload.new, retrieval = nil)
+    @model = model
+    @payload = payload
+    @retrieval = retrieval ||
+        Magma::Retrieval.new(model, nil, model.attributes.values, nil, 1, 100_000)
+  end
+
+  def tsv_dump(options = {})
+    gzip = options[:gzip]
+    file_dir = options[:file_dir]
+
+    if !file_dir
+      time = Time.now
+      # default file_dir format e.g. "/var/tmp/project_model_2018-2-13_dump.tsv"
+      file_dir = "/var/tmp/#{@model.project_name.to_s}_#{@model.model_name.to_s}_#{time.year.to_s}-#{time.month.to_s}-#{time.day.to_s}_dump.tsv"
```

Comment:
We shouldn't hard-code any file locations since we can't assume /var/tmp exists or is writeable